docs/configuration.md

```diff
@@ -3134,3 +3134,111 @@ The stage level scheduling feature allows users to specify task and executor res
 This is only available for the RDD API in Scala, Java, and Python.  It is available on YARN and Kubernetes when dynamic allocation is enabled. See the [YARN](running-on-yarn.html#stage-level-scheduling-overview) page or [Kubernetes](running-on-kubernetes.html#stage-level-scheduling-overview) page for more implementation details.
 
 See the `RDD.withResources` and `ResourceProfileBuilder` API's for using this feature. The current implementation acquires new executors for each `ResourceProfile`  created and currently has to be an exact match. Spark does not try to fit tasks into an executor that require a different ResourceProfile than the executor was created with. Executors that are not in use will idle timeout with the dynamic allocation logic. The default configuration for this feature is to only allow one ResourceProfile per stage. If the user associates more then 1 ResourceProfile to an RDD, Spark will throw an exception by default. See config `spark.scheduler.resource.profileMergeConflicts` to control that behavior. The current merge strategy Spark implements when `spark.scheduler.resource.profileMergeConflicts` is enabled is a simple max of each resource within the conflicting ResourceProfiles. Spark will create a new ResourceProfile with the max of each of the resources.
+
+# Push-based shuffle overview
+
+Push based shuffle helps improve the reliability and performance of spark shuffle. It takes a best-effort approach to push the shuffle blocks generated by the map tasks to remote shuffle services to be merged per shuffle partition. Reduce tasks fetch a combination of merged shuffle partitions and original shuffle blocks as their input data, resulting in converting small random disk reads by shuffle services into large sequential reads. Possibility of better data locality for reduce tasks additionally helps minimize network IO.
+
+<p> <b> Currently push-based shuffle is only supported for Spark on YARN with external shuffle service. </b></p>
+
+### Shuffle server side configuration options
+
+<table class="table">
+<tr><th>Property Name</th><th>Default</th><th>Meaning</th><th>Since Version</th></tr>
+<tr>
+  <td><code>spark.shuffle.push.server.mergedShuffleFileManagerImpl</code></td>
+  <td>
+    <code>org.apache.spark.network.shuffle.ExternalBlockHandler$NoOpMergedShuffleFileManager</code>
```

Comment:
It is hacky to contain `$` in the configuration value. Also, the config value is so long that it affects the readability of the doc:
![image](https://user-images.githubusercontent.com/1097932/128316212-16e704e4-b9c2-4ad5-80ec-1822387185a0.png)

Reply:
Is it possible to move `NoOpMergedShuffleFileManager` under `org.apache.spark.network.shuffle`?

Reply:
Good suggestion! Would it cause issues while copy paste'ing the config for users?

Reply:
> Would it cause issues while copy paste'ing the config for users

Yes it could be.

Reply:
Any suggestions?

Reply:
Shall we move NoOpMergedShuffleFileManager under org.apache.spark.network.shuffle? So that the value becomes
`org.apache.spark.network.shuffle.NoOpMergedShuffleFileManager`

Reply:
+CC @venkata91 Did you get a chance to raise a PR to refactor and pull this inner class out ?

Reply:
@mridulm Yeah I haven't tried that yet. But still the config key names are quite long, it would still not make the readability issue go away. Shouldn't this be handled at the CSS layer? Thoughts?

Reply:
Ok. I tried changing CSS for `td` to `word-wrap` works fine, but it also changes the other tables. We would still have the issue of `$` for the config value which is an issue. Let me file another PR to handle that.

Reply:
PR for pulling out `NoOpMergedShuffleFileManager` - https://github.com/apache/spark/pull/33688